### ISSUE TITLE
groups (e.g. `alt`) with `else` cannot be rendered in parallel

### ISSUE THREAD

**Issue by da-kami** [plantuml/plantuml#269]
This renders correctly:
```plantuml
@startuml
!pragma teoz true
    opt
        A->B: message
    end
    & opt
        C->D: message
    end

    alt
        A->B: message
    end
    & alt
        C->D: message
    end

    alt
        A->B: message
    else default
        B->B: other
    end
    alt
        C->D: message
    else default
        D->D: other
    end

@enduml
```

![image](https://user-images.githubusercontent.com/5557790/67444659-bd4b6a00-f655-11e9-8c4f-cc410ec65878.png)

Once the `alt` with `else` is parallelised it throws a `NullPointer` exception. 
```plantuml
@startuml
!pragma teoz true
    opt
        A->B: message
    end
    & opt
        C->D: message
    end

    alt
        A->B: message
    end
    & alt
        C->D: message
    end

    alt
        A->B: message
    else default
        B->B: other
    end
    & alt
        C->D: message
    else default
        D->D: other
    end

@enduml
```

![image](https://user-images.githubusercontent.com/5557790/67444692-e409a080-f655-11e9-8dff-ad4af43c2233.png)

**Comment by jimnelson372** [comment]
@da-kami and @The-Lum, 

I was able to track down the cause of the exception and find a fix.   It required the changes I made for #591 to properly position the else header line and message, but it works well.   

First, here is what my test environment generates now when running your exact script:

![testteozaltparallel_005](https://github.com/plantuml/plantuml/assets/66284321/2c9438fe-8804-40a8-b88a-120e39136271)

The alt/else blocks are now in parallel, but obviously there is overlap on the two groups.  That's because the internal sides of each group do not push push away other lifelines or events, but merely surround the events they contain.  So in this case the `Alt` groups overlap.  Even the gap in your original parallel `Opt` and `Alt` blocks only were separated due to the needed width of `B->B : other` self message further down in your script.   (I'm somewhat new to PlantUML and the community so I don't know if there are discussions about groups altering (or not altering) the layout.  Though without the parallel (&) feature in Teoz, this wouldn't have been an issue.)

To remove the overlap, I added a hidden message between B and C with sufficient width:

`    B -[hidden]-> C: "              "`

This results in:

![testteozaltparallel_005](https://github.com/plantuml/plantuml/assets/66284321/bcb444fb-b5d1-4c10-99b9-77d40eca169d)

I believe that is what you were looking for!   I think it looks good.   

Let me know if there is anything that looks wrong about this.  If it's all okay, I'll create a pull request to add it to the code.

@The-Lum, is the group overlap expected shown in the first generated diagram expected, or does that suggest a problem?

Regards,

Jim Nelson
